yaks/lib/yaks/mapper/attribute.rb

```diff
@@ -2,22 +2,23 @@ module Yaks
   class Mapper
     class Attribute
       extend Forwardable, Util
-      include Util
-      include Attribs.new(:name, :block, options: {}.freeze)
+      include Attribs.new(:name, :block, if: true), Util
 
-      def self.create(*args, &block)
-        args, options = extract_options(args)
-        new(name: args.first, options: options, block: block)
+      def self.create(name, options = {}, &block)
+        options[:name] = name
+        options[:block] = block
```

Comment:
You're mutating an object passed in by the caller. That's not a good idea. Better to use something like

``` ruby
options = options.merge(...)
```